tq/basic_download.go

```diff
@@ -42,52 +43,80 @@ func (a *basicDownloadAdapter) WorkerEnding(workerNum int, ctx interface{}) {
 }
 
 func (a *basicDownloadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb ProgressCallback, authOkFunc func()) error {
-	f, fromByte, hashSoFar, err := a.checkResumeDownload(t)
+	// Reserve a temporary filename. We need to make sure nobody operates on the file simultaneously with us.
+	f, err := ioutil.TempFile(a.tempDir(), t.Oid)
 	if err != nil {
 		return err
 	}
-	return a.download(t, cb, authOkFunc, f, fromByte, hashSoFar)
-}
+	// This will delete temp file if:
+	// - we failed to fully download file and move it to final location including the case when final location already
+	//   exists because other parallel git-lfs processes downloaded file
+	// - we also failed to move it to a partially-downloaded location
+	defer os.Remove(f.Name())
```

Comment:
You can put more than one statement into a single defer by using construct:
```
defer func() {
  ...
  ...
}()
```
This with gurantee the order of operations. I'd close file before deleting it.

Reply:
Fixed